service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -63,6 +61,14 @@ func (r *StateGetter) GetDesiredState(ctx context.Context, obj interface{}) ([]*
 		}
 	}
 
+	var ingressControllerLegacy bool
+	{
+		ingressControllerLegacy = true
+		if r.provider == "azure" {
+			ingressControllerLegacy = false
+		}
+	}
```

Comment:
The naming of the legacy flag now gets a bit confusing. As this is the legacy branch of cluster-operator but the flag is now false for Azure.

AIUI this is needed because ingress controller becomes optional on Azure. Can you add a comment explaining that?

Reply:
Makes sense, done